### ISSUE TITLE
inherited Awake methods don't call.

### ISSUE THREAD

**Issue by FirstGearGames** [FirstGearGames/FishNet#40]
```
public class A : B
{

}

public class B : NetworkBehaviour
{
    private void Awake()
    {
       UserLogic();
    }
}
```
`class B` unexpected does not run Awake.

**Comment by FirstGearGames** [comment]
Resolved in 1.4.8.

**Comment by martinzudaire** [comment]
Errors with new NetworkBehaviourProcessor.cs file:

```
ArgumentException: An item with the same key has already been added. Key: 0
System.Collections.Generic.Dictionary`2[TKey,TValue].TryInsert (TKey key, TValue value, System.Collections.Generic.InsertionBehavior behavior) (at <6073cf49ed704e958b8a66d540dea948>:0)
System.Collections.Generic.Dictionary`2[TKey,TValue].Add (TKey key, TValue value) (at <6073cf49ed704e958b8a66d540dea948>:0)
FishNet.Object.NetworkBehaviour.RegisterSyncType (FishNet.Object.Synchronizing.Internal.SyncBase sb, System.UInt32 index) (at Assets/FishNet/Runtime/Object/NetworkBehaviour.SyncTypes.cs:91)
FishNet.Object.Synchronizing.Internal.SyncBase.InitializeInstance (FishNet.Object.NetworkBehaviour nb, System.UInt32 syncIndex, FishNet.Object.Synchronizing.WritePermission writePermissions, FishNet.Object.Synchronizing.ReadPermission readPermissions, System.Single tickRate, FishNet.Transporting.Channel channel, System.Boolean isSyncObject) (at Assets/FishNet/Runtime/Object/Synchronizing/SyncBase.cs:83)
FishNet.Object.Synchronizing.SyncVar`1[T]..ctor (FishNet.Object.NetworkBehaviour nb, System.UInt32 syncIndex, FishNet.Object.Synchronizing.WritePermission writePermission, FishNet.Object.Synchronizing.ReadPermission readPermission, System.Single sendRate, FishNet.Transporting.Channel channel, Swarm.Prototype.Gun value) (at Assets/FishNet/Runtime/Object/Synchronizing/SyncVar.cs:70)
Swarm.Prototype.Projectile.NetworkInitialize___Early () <0x2578028b580 + 0x000fa> in <e124d21e7e4943e9aa4c7788efce3cb2>:0
Swarm.Prototype.Projectile.Awake () (at Assets/_Project/Scripts/Guns/Projectiles/Projectile.cs:71)
Swarm.Prototype.HitscanBullet.Awake () (at <e124d21e7e4943e9aa4c7788efce3cb2>:0)
UnityEngine.Object:Instantiate(GameObject, Vector3, Quaternion)
Swarm.Prototype.Gun:RpcLogic___RpcFire_2166136261() (at Assets/_Project/Scripts/Guns/Gun.cs:277)
Swarm.Prototype.Gun:RpcReader___Server_RpcFire_2166136261(PooledReader, Channel, NetworkConnection)
FishNet.Object.NetworkBehaviour:OnServerRpc(PooledReader, NetworkConnection, Channel) (at Assets/FishNet/Runtime/Object/NetworkBehaviour.RPCs.cs:153)
FishNet.Managing.Server.ServerObjects:ParseServerRpc(PooledReader, NetworkConnection, Channel) (at Assets/FishNet/Runtime/Managing/Server/Object/ServerObjects.Parsing.cs:39)
FishNet.Managing.Server.ServerManager:ParseReceived(ServerReceivedDataArgs) (at Assets/FishNet/Runtime/Managing/Server/ServerManager.cs:503)
FishNet.Managing.Server.ServerManager:Transport_OnServerReceivedData(ServerReceivedDataArgs) (at Assets/FishNet/Runtime/Managing/Server/ServerManager.cs:400)
FishNet.Transporting.Tugboat.Tugboat:HandleServerReceivedDataArgs(ServerReceivedDataArgs) (at Assets/FishNet/Runtime/Transporting/Transports/Tugboat/Tugboat.cs:233)
FishNet.Transporting.Tugboat.Server.ServerSocket:IterateIncoming() (at Assets/FishNet/Runtime/Transporting/Transports/Tugboat/Core/ServerSocket.cs:558)
FishNet.Transporting.Tugboat.Tugboat:IterateIncoming(Boolean) (at Assets/FishNet/Runtime/Transporting/Transports/Tugboat/Tugboat.cs:192)
FishNet.Managing.Transporting.TransportManager:IterateIncoming(Boolean) (at Assets/FishNet/Runtime/Managing/Transporting/TransportManager.cs:320)
FishNet.Managing.Timing.TimeManager:TryIterateData(Boolean) (at Assets/FishNet/Runtime/Managing/Timing/TimeManager.cs:815)
FishNet.Managing.Timing.TimeManager:IncreaseTick() (at Assets/FishNet/Runtime/Managing/Timing/TimeManager.cs:610)
FishNet.Managing.Timing.TimeManager:TickUpdate() (at Assets/FishNet/Runtime/Managing/Timing/TimeManager.cs:312)
FishNet.Transporting.NetworkReaderLoop:Update() (at Assets/FishNet/Runtime/Transporting/NetworkReaderLoop.cs:28)
```
